### DISCUSSION TITLE
Add "comment and resolve" button for note discussions?

### DISCUSSION THREAD

**Discussion by westnordost** [streetcomplete/StreetComplete#4700] [Ideas]
As StreetComplete has been becoming more powerful over time, particularly because of the overlays and because of the fact that over time "other answers" of common situations are being added where the user would otherwise have selected "can't answer" in the past.

So, there will always be a number of notes that would now be solvable on-site with StreetComplete if it would not block any quest and any editing via overlays due to an open note. If the app would allow to close a note discussion, maybe with a due warning that explains in which situations it is OK to close a note, these situations could be solved.

So, do you think it would make sense to add this functionality? Pro and Contra I see... anything more to add?

**Pro**:
- some notes become solvable within StreetComplete, in particular those created with earlier versions of the app

**Contra**:
- only partly solving the use case described in #4687: If the note is unrelated to the edit one wants to do (e.g. note complains about turn restrictions but one wants to edit the road surface), it would not be OK to close that note just to be able to edit the road surface
- (thus) danger of users unduly or accidentally closing notes
- slightly more involved UI

**Comment by matkoniecz** [comment]
part of slightly more involved UI is necessity to explain to users when note should be marked as closed (after fixing data, not after providing info necessary to solve them)

**Comment by mnalis** [comment]
- I definitely do think that SC has advanced enough that it can solve situations that it couldn't previously (e.g. especially with overlays; but a few quests have been improved too), as I noted e.g. in https://github.com/streetcomplete/StreetComplete/issues/4529   

- I however also agree with @matkoniecz that notes should be closed only **after** their issue has been solved. And **not** as **_prerequisite_** for solving it. Otherwise it is quite likely (in my experience) that notes would be closed, _without_ actually solving the issue (_especially_ if it is as easy as clicking a button), which I find as a bigger problem than few quests which are not shown in SC (but could be shown if note was not there).

Thus I would not be too happy with such simple _"Close note"_ functionality in SC, as seems to be suggested. 

---

What would be good solution for me (although I'm quite aware it is more complex and thus possibly outside of the scope of this idea):

- Allow the Overlays to always edit the data, regardless of the presence of Notes on that data.
- After the Overlay edit is confirmed, check if there was the Note on that data. If there is a Note on it, either:
   - **option (a)**:  show the user whole note (with additional comments, pictures etc) and asked them if all of that is now solved (and closed the note if user answers "yes"). That gives move control to the user and closes more notes, but is more risky if the user is newbie and does not really fully comprehend what was the original problem. 
   - **option (b)**:  show the user whole note (with additional comments, pictures etc) and asked them to describe how they've solved it, and that use that text to add a comment to the note (instead of closing it). That is a little more work for the user, but reduces the risk of wrongly closed notes, and adds possibility that users watching the notes will verify and close them if correct (or ask additional questions if they deem it is not solved yet).
   -  **option (c)**: without asking the user anything, add automatic hardcoded-text comment to the Note (without closing it), saying that _"changeset #xxxx might (or might not) have solved that note, so it should be verified and either closed, or extra questions added"_.  That is lowest amount of work for on-the-ground user, and while not ideal from Note creator perspective, at least does allow editing the data and makes Note watchers aware of changed situation.

I would prefer option (b), but any of them would IMHO be an improvement over current situation.

<details>

I do recognize even that would only solve majority of the issues, not all of them (e.g.it would help the overlays case as in #4529, but not with [when POI and note have the same position #4687](https://github.com/streetcomplete/StreetComplete/issues/4687) - for that one, something like previously suggested [Allow disabling Note quest #3532](https://github.com/streetcomplete/StreetComplete/issues/3532) would be needed).

</details>

**Comment by rhhsm** [comment]
A bit of user experience that might help: many notes are of the "there is a shop here" kind. Sometimes I solve them at home by searching the information in the note (such as the shop name) and if I can find confirmatory information on Internet (such as the shop's website that confirms its address), I add the shop using iD. If not, I go to survey and see if I can find the shop on the ground. If so, I add it using the shop overlay. Here I would have liked to be able to close the note. However, so far I just added "added the shop" to the note (option b), and closed it when I got back home. That works fine for me, and I understand that this is preferable to allowing inexperienced SC users to close notes without fully understanding the problem highlighted in the note. Optionally, SC users could be given the privilege of being allowed to close notes as a reward after reaching a certain experience level (number of stars collected).

**Reply by mnalis** [reply]
I don't really think that it is good idea to implement technical measures that will prevent new users from using some functionality just because they haven't reached some high enough score. 
I think such **technical blockade would only needlessly aggravate them** and make them regret the SC, or even worse, make them submit non-surveyed information just to get above some arbitrary threshold and get a functionality they want.  
Gamification is all fun and games until someone gets hurt! :smiling_face_with_tear:  I think it should remain tied only to "useless" rewards (i.e. links to sites they could find with search engine themselves etc)

If such functionality needs to be implemented but hidden from newbies by default, IMHO better add it in preference behind some _"expert mode"_ checkbox which displays a warning and asks a confirmation (like problematic quests do). But we've been over that idea (https://github.com/streetcomplete/StreetComplete/discussions/3003) and the resolution generally was that such "expert mode" is better belonging in some fork - which is what _[SCEE](https://wiki.openstreetmap.org/wiki/SCEE ) (StreetComplete Expert Edition)_ has become. But maybe for this specific functionality such checkbox would be OK method?

Incidentally, **SCEE has since implemented that functionality for closing notes**.  When commenting on a note, there is checkbox which one can check to close the note. But one shouldn't do that unless the issue was not solved first, obviously. It is OK solution for SCEE where there is just small print saying so, because SCEE is geared towards more advanced users. But I don't think such solution would work well for regular StreetComplete. Maybe if it has extra warning/confirmation popup on each attempt to close the note? 

But all things considered, and given the existence of SCEE, I think it is probably better if closing notes were **not** implemented in regular StreetComplete. Too much potential for closing by mistake, and only more advanced users would benefit. Target StreetComplete demographics (newbie users that don't even need to have technical knowledge of what Notes are, much less that they can have open/close status -- [ObXKCD](https://xkcd.com/2501/)) can leave comments on the note that it is resolved, and someone else more advanced can close them.
